de.dlr.sc.virsat.project.test/src/de/dlr/sc/virsat/project/structure/command/CreateRemoveSeiWithFileStructureCommandTest.java

```diff
@@ -0,0 +1,178 @@
+/*******************************************************************************
+ * Copyright (c) 2020 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.project.structure.command;
+
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.hasItems;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.emf.common.command.Command;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.transaction.RecordingCommand;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import de.dlr.sc.virsat.model.dvlm.Repository;
+import de.dlr.sc.virsat.model.dvlm.structural.StructuralElement;
+import de.dlr.sc.virsat.model.dvlm.structural.StructuralElementInstance;
+import de.dlr.sc.virsat.model.dvlm.structural.StructuralFactory;
+import de.dlr.sc.virsat.project.test.AProjectTestCase;
+
+public class CreateRemoveSeiWithFileStructureCommandTest extends AProjectTestCase {
+	
+	StructuralElement se;
+	
+	StructuralElementInstance sei1;
+	StructuralElementInstance sei2;
+	StructuralElementInstance sei3;
+	
+	IFolder seiFolder1;
+	IFolder seiFolder2;
+	IFolder seiFolder3;
+	
+	Resource resourceSei1;
+	Resource resourceSei2;
+	Resource resourceSei3;
+	
+	@Before
+	public void setUp() throws CoreException {
+		super.setUp();
+		addEditingDomainAndRepository();
+
+		editingDomain.getCommandStack().execute(new RecordingCommand(editingDomain) {
```

Comment:
Can we use the `executeAsCommand(...)` here?

Reply:
Absolutely! Forgot that we provided this back in the days...